src/config.c

```diff
@@ -91,6 +92,26 @@ static struct cgroup_string_list *template_files;
 /* Needed for the type while mounting cgroupfs. */
 #define CGROUP_FILESYSTEM "cgroup"
 
+#ifdef WITH_SYSTEMD
+/* Directory that holds dynamically created internal libcgroup files */
+static const char * const systemd_def_cgrp_file_dir = "/var/run/libcgroup/";
+
+/* File that stores the relative delegated systemd cgroup path */
+static const char * const systemd_default_cgroup_file = "/var/run/libcgroup/systemd";
+
+/* Lock that protects systemd_default_cgroup_file */
+static pthread_rwlock_t systemd_default_cgroup_lock = PTHREAD_RWLOCK_INITIALIZER;
+
+/*
+ * cgroup_systemd_opts is dynamically allocated, hence having head
+ * and tail helps in traversing.
+ */
+static struct cgroup_systemd_opts *cgroup_systemd_opts_head;
```

Comment:
Unless I'm missing something, I think we need to initialize these to NULL.

Reply:
@drakenclimber Thank you so much for the review.
Isn't `static` storage initialized to a null pointer unless initialized explicitly?
```
$ echo -e "struct { int num;}a;\n struct a *foo;\n int main() {\n if (foo) { return 1;}\n return 0;\n}" | gcc -x c - -o -
$ echo $?
0
```